pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -643,17 +654,193 @@ func TestUserSignupWithExistingMUROK(t *testing.T) {
 	err = r.client.Get(context.TODO(), key, instance)
 	require.NoError(t, err)
 
+	require.NotEmpty(t, instance.Status.CompliantUsername)
+
+	// Confirm that the mur exists
+	mur = &v1alpha1.MasterUserRecord{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: operatorNamespace, Name: instance.Status.CompliantUsername}, mur)
+	require.NoError(t, err)
+	require.Equal(t, instance.Name, mur.Labels[v1alpha1.MasterUserRecordUserIDLabelKey])
+
 	var cond *v1alpha1.Condition
 	for _, condition := range instance.Status.Conditions {
 		if condition.Type == v1alpha1.UserSignupComplete {
 			cond = &condition
 		}
 	}
 
+	require.Equal(t, mur.Name, instance.Status.CompliantUsername)
 	require.NotNil(t, cond)
 	require.Equal(t, v1.ConditionTrue, cond.Status)
 }
 
+func TestUserSignupWithInvalidNameNotOK(t *testing.T) {
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      uuid.NewV4().String(),
+			Namespace: operatorNamespace,
+			UID:       types.UID(uuid.NewV4().String()),
+		},
+		Spec: v1alpha1.UserSignupSpec{
+			Username: "foo#bar@redhat.com",
+			Approved: false,
+		},
+	}
+
+	r, req, _ := prepareReconcile(t, userSignup.Name, userSignup, configMap(config.UserApprovalPolicyAutomatic))
+
+	createMemberCluster(r.client)
+	defer clearMemberClusters(r.client)
+
+	_, err := r.Reconcile(req)
+	require.Error(t, err)
+
+	key := types.NamespacedName{
+		Namespace: operatorNamespace,
+		Name:      userSignup.Name,
+	}
+	instance := &v1alpha1.UserSignup{}
+	err = r.client.Get(context.TODO(), key, instance)
+	require.NoError(t, err)
+
+	var cond *v1alpha1.Condition
+	for _, condition := range instance.Status.Conditions {
+		if condition.Type == v1alpha1.UserSignupComplete {
+			cond = &condition
+		}
+	}
+
+	require.NotNil(t, cond)
+	require.Equal(t, v1.ConditionFalse, cond.Status)
```

Comment:
Use the same check as in https://github.com/codeready-toolchain/host-operator/pull/93/files#diff-d7d6486f50f2c1cd16101f429e4ef843R601